include/Core/System/Window.hpp

```diff
@@ -10,6 +13,56 @@
 #include <string>
 #include <vili/node.hpp>
 
+namespace obe::events
+{
+    namespace Window
+    {
+        struct Position
+        {
+            unsigned int x;
+            unsigned int y;
+        };
+
+        struct Size
+        {
+            unsigned int width;
+            unsigned int height;
+        };
+
+        struct Moved
+        {
+            static constexpr std::string_view id = "Moved";
+            Position new_pos;
```

Comment:
As in https://github.com/ObEngine/ObEngine/blob/6ad69858c82733b798881cd9fe4b0a90e2af52f2/include/Core/System/Cursor.hpp#L20

A `position` and `previous_position` naming scheme would be more appropriate imo

Reply:
@Symei I have a Position new_pos and then Position previous_pos, Position is a struct that has unsigned int values for x,y. Are you saying remove the Position struct, and just have unsigned int x,y values vs a struct called Position, that contains x,y?

Reply:
You can keep the `Position` struct, just rename `new_pos` to `position` and `previous_pos` to `previous_position` and we're good to go :)